content/en/device-provisioning-service/client-library.md

```diff
@@ -50,49 +51,75 @@ void plgd_dps_shutdown(void);
 plgd_dps_context_t *plgd_dps_get_context(size_t device);
 
 /// @brief Set DPS manager callbacks.
-void plgd_dps_manager_init_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
-                                     void *on_change_data, oc_cloud_cb_t on_cloud_change_cb,
-                                     void *on_cloud_change_data);
+void plgd_dps_set_manager_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
+  void *on_change_data, oc_cloud_cb_t on_cloud_change_cb, void *on_cloud_change_data);
 
 /**
  * @brief Start DPS manager to provision device by given server.
  *
- * Setup context, handlers, dps server and start dps manager.
+ * Setup context, handlers, dps endpoint and start dps manager.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
  * @param endpoint endpoint of the provisioning server (cannot be NULL)
  * @return 0 on success
  * @return -1 on failure
  */
-int plgd_dps_manager_start(plgd_dps_context_t *ctx, const char *endpoint);
+int plgd_dps_manager_start(plgd_dps_context_t *ctx);
 
 /**
  * @brief Stop DPS manager.
  *
  * Deregister handlers, clear context, stop manager and close connection.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
  */
 void plgd_dps_manager_stop(plgd_dps_context_t *ctx);
 
 /**
- * @brief Clean-up and restart DPS provisioning on factory reset.
+ * @brief Restart DPS manager to provision device by given server.
+ *
+ * Setup context, handlers, dps endpoint and start dps manager.
+ *
+ * @param ctx dps context (cannot be NULL)
+ * @param endpoint endpoint of the provisioning server (cannot be NULL)
+ * @return 0 on success
+ * @return -1 on failure
+ */
+int plgd_dps_manager_restart(plgd_dps_context_t *ctx);
```

Comment:
What happens if the dps is uninitialized and I call it? What if it is initialized and I call it? What if there is a failure? Do I need to set address again? It's unclear in what situation should I use it and how it behaves.

Reply:
Currently it's just a convenience function that calls stops and start. I think it's enough just to reference the start and stop functions, which should be fully documented.